src/components/CsvUpload/CsvUpload.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useRef, useState } from 'react'
+import { FileRejection, useDropzone } from 'react-dropzone'
+import { HStack, Spacer, VStack } from '../ui/Stack/Stack'
+import { P } from '../ui/Typography/Text'
+import UploadCloud from '../../icons/UploadCloud'
+import classNames from 'classnames'
+import { IconButton } from '../Button'
+import { DataState, DataStateStatus } from '../DataState/DataState'
+import { FileSpreadsheet } from 'lucide-react'
+import CloseIcon from '../../icons/CloseIcon'
+
+const validateCsvFile = (file: File) => {
```

Comment:
I think a very liberal size limitation might be useful here as well since I think we have a size attribute available.
Something like 10mb is outrageously generous for a CSV of transactions, but might help limit against strange user misunderstaings.